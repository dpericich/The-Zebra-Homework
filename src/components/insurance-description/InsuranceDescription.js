import React from "react";

const InsuranceDescription = ({ name, details }) => {
  return (
    <div className="insurance-description">
      <div className="insurance-description__title">Why {name}?</div>
      <div className="insurance-description__details">
        <div className="insurance-description__text">{details}</div>
        <div className="insurance-description__disclaimer">
          Price shown is a soft quote based on the {name} insurance product in
          your state.
        </div>
      </div>
    </div>
  );
};

export default InsuranceDescription;
