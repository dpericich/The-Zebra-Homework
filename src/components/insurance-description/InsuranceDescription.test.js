import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import InsuranceDescription from "./InsuranceDescription";

describe("<InsuranceDescription />", () => {
  test("it renders without issue", () => {
    const component = shallow(<InsuranceDescription />);
    expect(toJSON(component)).toMatchSnapshot();
  });

  test("it renders correctly with name and description props passed in", () => {
    const name = "Fidelity";
    const description = "This is a great insurance for your banking";
    const component = shallow(
      <InsuranceDescription name={name} description={description} />
    );
    expect(toJSON(component)).toMatchSnapshot();
  });
});
