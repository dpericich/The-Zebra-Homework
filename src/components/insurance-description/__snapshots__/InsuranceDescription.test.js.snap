// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InsuranceDescription /> it renders correctly with name and description props passed in 1`] = `
<div
  className="insurance-description"
>
  <div
    className="insurance-description__title"
  >
    <ion-icon
      name="book-outline"
    />
     Why 
    Fidelity
    ?
  </div>
  <div
    className="insurance-description__details"
  >
    <div
      className="insurance-description__text"
    />
    <div
      className="insurance-description__disclaimer"
    >
      Price shown is a soft quote based on the 
      Fidelity
       insurance product in your state.
    </div>
  </div>
</div>
`;

exports[`<InsuranceDescription /> it renders without issue 1`] = `
<div
  className="insurance-description"
>
  <div
    className="insurance-description__title"
  >
    <ion-icon
      name="book-outline"
    />
     Why 
    ?
  </div>
  <div
    className="insurance-description__details"
  >
    <div
      className="insurance-description__text"
    />
    <div
      className="insurance-description__disclaimer"
    >
      Price shown is a soft quote based on the 
       insurance product in your state.
    </div>
  </div>
</div>
`;
